### PR TITLE
Feature/604

### LINKED ISSUE #604: Update template endpoints - Custom no-reply email
# Value Statement
**As** a business line
Add a no-reply email to the template
to allow for monitoring of custom email inboxes per business line

The current templates do not have the `no-reply` option in there, need to make sure that this is all updated all the way through and can accept the `no-reply` when sending to the template.

# Acceptance Criteria
**GIVEN** 
**WHEN** 
**THEN**

# Checklist
- [ ] Checklist item here

# Cross Functional Checklist

Testing:
  - [ ] integration test for API

Monitoring:
  - [ ] statsd metrics
  - [ ] datadog graphs
  - [ ] datadog alterts

Security:
  - [ ] auth permissions
  - [ ] JsonSchema

Transform requests to business models in our API
- [ ] Marshmallow model serialization / deserialization

Documentation:
  - [ ] Readmes
  - [ ] Project specific
  - [ ] Support
  - [ ] Openapi
  - [ ] ADR
  - [ ] Tech Debt wall

Mountebank
- [ ] API - create mocks for any external services calls

Postman
- [ ] Update collection whenever we change our API

# Assumptions
-

# Additional Info/Resources
-

# Out of Scope
-

# Open Questions
-

### LINKED ISSUE #606: Update code to support no-reply from email by template
## Acceptance Criteria

**GIVEN** 
**WHEN**
**THEN**

## Checklist

-   [ ] Checklist item here

## Assumptions

-

## Additional Info/Resources

-

## Out of Scope

-

## Open Questions

-

## Cross Functional Checklist

Testing:

-   [ ] integration tests for API
-   [ ] Cypress (e2e) for UI

Monitoring:

-   [ ] statsd metrics
-   [ ] datadog graphs
-   [ ] datadog alterts

Security:

-   [ ] auth permissions and roles
-   [ ] Move API enpdoints to new user token auth

Marshmallow

-   [ ] Model serialization / deserialization

Documentation:

-   [ ] Readmes
-   [ ] Project specific
-   [ ] Support
-   [ ] Openapi
-   [ ] ADR
-   [ ] Tech Debt wall

Mountebank

-   [ ] Mock API enpoints including error scenarios

Accessibility

-   [ ] E2E & Testrail

Security:

-   [ ] auth permissions
-   [ ] JsonSchema

Postman

-   [ ] Update collection

### PR DESCRIPTION
**Pull request by jessecanderson**
### Fixes #604 

This updates the `service/{{service-id}}/template/{{email-template-id}}` endpoint to allow us to update the template with a custom `reply_to_email` value. 

### Steps to Test
- [ ] Run the API
- [ ] Open Postman Collection
- [ ] Select a template to test with. Example one could be Service ID of `d6aa2c68-a2d9-4437-ab19-3ae8eb202553` and  Template ID of `618185c6-3636-49cd-b7d2-6f6f5eb3bdde`
- [ ] Select the `/service/template/update EMAIL template` message
- [ ] Insert the body to update the Template (see below for example message`
- [ ] Under `/service/template/for EMAIL` select the `get EMAIL template` and validate the returned template now has a `reply_to_email` assigned value

POST message body for updating the template:
```
{
    
    "reply_to_email": "bev_is_great@email.com"
    
}
```

### CODE REVIEW

File: tests/app/dao/test_templates_dao.py [hunk 1]
```diff
@@ -96,6 +96,28 @@ def test_update_template(sample_service, sample_user):
...
+    }
+    template = Template(**data)
+    dao_create_template(template)
+    created = dao_get_all_templates_for_service(sample_service.id)[0]
+    assert created.name == 'Sample Template'
+    assert created.reply_to_email == "some@one.com"
+
+    created.name = 'new name'
+    created.reply_to_email = "testing@change.com"
+    dao_update_template(created)
```
Comment (kalbfled) [inline tests/app/dao/test_templates_dao.py:116]: This tests the function that the API endpoint code calls, but it doesn't test the API endpoint code, contrary to your test name.  You should test both.  You also haven't tested that the new TemplateHistory row gets created as expected.

File: tests/app/dao/test_templates_dao.py [hunk 2]
```diff
@@ -96,6 +96,28 @@ def test_update_template(sample_service, sample_user):
...
+    assert dao_get_all_templates_for_service(sample_service.id)[0].name == 'new name'
+    assert dao_get_all_templates_for_service(sample_service.id)[0].reply_to_email == "testing@change.com"
+
+
```
Comment (kalbfled) [inline tests/app/dao/test_templates_dao.py:120]: I recommend writing a test that updates multiple fields at once.

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: tests/app/dao/test_templates_dao.py (hunk 1 above)
Comment (jessecanderson) [reply tests/app/dao/test_templates_dao.py:116]: I'm pretty sure this test (which I did base off one of the other tests) was not to test the API endpoint but being inside the `tests/app/dao` section was to only test the function that I created to update the template. I will see if I can locate the API tests and update them. I am also in the process of consolidating all the `update template` methods into a single one per your suggestion so these will have to change to accomidate that test of the DOA method updates anyways.

File: tests/app/dao/test_templates_dao.py (hunk 2 above)
Comment (jessecanderson) [reply tests/app/dao/test_templates_dao.py:120]: This should be able to be tested after all the updates are completed and will be located with any other API tests and maybe not specifically the DAO tests.

File: tests/app/dao/test_templates_dao.py [hunk 3]
```diff
@@ -96,6 +96,28 @@ def test_update_template(sample_service, sample_user):
     assert dao_get_all_templates_for_service(sample_service.id)[0].name == 'new name'
 
 
+def test_update_template_with_reply_to_email(sample_service, sample_user):
+    data = {
+        'name': 'Sample Template',
+        'template_type': "sms",
```
[the remark this answers is not in the document]
Comment (jessecanderson) [reply tests/app/dao/test_templates_dao.py]: Test was removed as not needed.

File: tests/app/dao/test_templates_dao.py (hunk 1 above)
Comment (jessecanderson) [reply tests/app/dao/test_templates_dao.py]: Test ended up not really being needed and has been removed.

File: tests/app/dao/test_templates_dao.py (hunk 2 above)
Comment (jessecanderson) [reply tests/app/dao/test_templates_dao.py]: Test was not needed and has been removed. Multiple update functionality is part of the endpoint now as I misunderstood what was going on earlier in the checks.

File: app/template/rest.py [hunk 4]
```diff
@@ -142,6 +143,10 @@ def update_template(service_id, template_id):
     if data.get('redact_personalisation') is True:
         return redact_template(fetched_template, data)
 
+    if "reply_to_email" in data:
+        updated = dao_update_template_reply_to_email(template_id=template_id, reply_to_email=data.get("reply_to_email"))
+        return jsonify(data=template_schema.dump(updated).data), 200
```
[the remark this answers is not in the document]
Comment (jessecanderson) [reply app/template/rest.py]: Ended up removing this and reverting back to original logic. We may want to investigate why the `reply_to` check stops the overall flow of this update and forces the template to update the version information as well.

File: app/dao/templates_dao.py [hunk 5]
```diff
@@ -84,6 +84,38 @@ def dao_update_template_reply_to(template_id, reply_to):
...
 
 
+@transactional
+def dao_update_template_reply_to_email(template_id, reply_to_email):
+    Template.query.filter_by(id=template_id).update(
+        {"reply_to_email": reply_to_email,
+         "updated_at": datetime.utcnow(),
+         "version": Template.version + 1,
+         }
+    )
```
[the remark this answers is not in the document]
Comment (jessecanderson) [reply app/dao/templates_dao.py]: Updated the formatting after removing my changes.

File: app/template/rest.py [hunk 6]
```diff
@@ -142,6 +143,10 @@ def update_template(service_id, template_id):
     if data.get('redact_personalisation') is True:
```
[the remark this answers is not in the document]
Comment (jessecanderson) [reply app/template/rest.py]: I didn't update this one specifically as I didn't write this line of code and wasn't 100% sure how this change would effect it. We can discuss in more detail or put in a tech debt story to update this.

**Review by kalbfled — APPROVED** [review]
I tested locally with Postman, and it seems to work.  Please squash your commits before merging.

**Pull request merged**